File: app/src/main/java/com/eugene_poroshin/money_manager/categories/CategoryViewModel.java
package com.eugene_poroshin.money_manager.categories;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.eugene_poroshin.money_manager.repo.Repository;
import com.eugene_poroshin.money_manager.repo.database.CategoryEntity;

import java.util.List;

public class CategoryViewModel extends AndroidViewModel {

    private Repository repository;
    private LiveData<List<CategoryEntity>> allCategoriesLiveData;
    private LiveData<List<String>> allCategoryNamesLiveData;

    public CategoryViewModel(@NonNull Application application) {
        super(application);
        repository = new Repository(application);
        allCategoriesLiveData = repository.getAllCategories();
        allCategoryNamesLiveData = repository.getCategoryNames();
    }

    public LiveData<List<CategoryEntity>> getLiveDataCategories() {
        return allCategoriesLiveData;
    }

    public LiveData<List<String>> getLiveDataCategoryNames() {
        return allCategoryNamesLiveData;
    }

    public void insert(CategoryEntity category) {
        repository.insert(category);
    }

    public void update(CategoryEntity category) {
        repository.update(category);
    }

    public void delete(CategoryEntity category) {
        repository.delete(category);
    }
}

File: app/src/main/java/com/eugene_poroshin/money_manager/ui/categories/CategoryViewModel.kt
package com.eugene_poroshin.money_manager.ui.categories

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.eugene_poroshin.money_manager.repo.CategoryRepository
import com.eugene_poroshin.money_manager.repo.database.CategoryEntity
import kotlinx.coroutines.launch

class CategoryViewModel(private val categoryRepository: CategoryRepository) : ViewModel() {

    val liveDataCategories: LiveData<List<CategoryEntity>> = categoryRepository.allCategories

    fun insert(category: CategoryEntity) = viewModelScope.launch {
        categoryRepository.insert(category)
    }

    fun update(category: CategoryEntity) = viewModelScope.launch {
        categoryRepository.update(category)
    }

    fun delete(category: CategoryEntity) = viewModelScope.launch {
        categoryRepository.delete(category)
    }
}